includes/reader-revenue/woocommerce/class-woocommerce-cover-fees.php

```diff
@@ -0,0 +1,233 @@
+					<?php
+						echo esc_html(
+							sprintf(
+								// Translators: %s is the transaction fee, as percentage with static portion (e.g. 2% + $0.3), %s is the site title.
+								__( 'Cover Stripe’s %1$s transaction fee, so that %2$s receives 100%% of your payment.', 'newspack-plugin' ),
+								self::get_fee_human_readable_value(),
+								get_option( 'blogname' )
+							)
+						);
+					?>
+
+				</label>
+			</p>
+		<?php
+		$desc .= ob_get_clean();
+		return $desc;
+	}
+
+	/**
+	 * Amend the price markup so it's possible to manipulate it via JS.
+	 *
+	 * @param string $html  Price HTML.
+	 * @param string $price Price.
+	 */
+	public static function amend_price_markup( $html, $price ) {
+		if ( ! self::should_allow_covering_fees() ) {
+			return $html;
+		}
+		return str_replace( $price, '<span id="' . self::PRICE_ELEMENT_ID . '">' . $price . '</span>', $html );
+	}
+
+	/**
+	 * Print the checkout helper JS script.
+	 */
+	public static function print_checkout_helper_script() {
+		if ( ! self::should_allow_covering_fees() ) {
+			return;
+		}
+		$handler = 'newspack-wc-modal-checkout-helper';
+		wp_register_script( $handler, '', [], false, [ 'in_footer' => true ] ); // phpcs:ignore WordPress.WP.EnqueuedResourceParameters.NoExplicitVersion
+		wp_enqueue_script( $handler );
+		$original_price          = WC()->cart->total;
+		$price_with_fee          = number_format( self::get_total_with_fee( $original_price ), 2 );
+		$has_coupons_available   = WC()->cart->get_coupon_discount_totals();
+		$coupons_handling_script = '';
+		if ( \wc_coupons_enabled() ) {
+			// Handle an edge case where the price was updated in the UI, and then a coupon was applied.
+			// In this case, the price has to be reverted to the original value, since covering fees
+			// is not supported with coupons.
+			$coupons_handling_script = 'setInterval(function(){
+				if(document.querySelector(".woocommerce-remove-coupon")){
+					document.getElementById( "' . self::PRICE_ELEMENT_ID . '" ).textContent =  "' . $original_price . '";
+				}
+			}, 1000);';
+		}
+		wp_add_inline_script(
+			$handler,
+			'function newspackHandleCoverFees(inputEl) {
+				document.getElementById( "' . self::PRICE_ELEMENT_ID . '" ).textContent = inputEl.checked ? "' . $price_with_fee . '" : "' . $original_price . '";
+			};' . $coupons_handling_script
+		);
+	}
+
+	/**
+	 * Get the fee multiplier value.
+	 */
+	private static function get_stripe_fee_multiplier_value() {
+		return floatval( get_option( 'newspack_blocks_donate_fee_multiplier', '2.9' ) );
+	}
+
+	/**
+	 * Get the fee static portion value.
+	 */
+	private static function get_stripe_fee_static_value() {
+		return floatval( get_option( 'newspack_blocks_donate_fee_static', '0.3' ) );
+	}
+
+	/**
+	 * Get the fee human-redable value.
+	 */
+	private static function get_fee_human_readable_value() {
+		$fee_static = sprintf( get_woocommerce_price_format(), get_woocommerce_currency_symbol(), self::get_stripe_fee_static_value() );
```

Comment:
How about using `wc_price()` so it uses all WC logic for price rendering? Using it will render $0.50 instead of $0.5 and theoretically handle any other exceptions a different currency may have.

Reply:
Good idea, done in 8c2f42972780c78ce97c31de2732b7412290710e